README.md

```diff
@@ -7,16 +7,20 @@ the idea behind this is sort of like a scripted version of [Borksy](https://ayol
 ## how
 
 1. run `npm i`
-1. copy-paste your gamedata into `./input/gamedata.txt`
+	* optionally add path to your local bitsy-hacks repo:
+	`npm i ../bitsy-hacks`
```

Comment:
i tried using npm link but then i noticed it reverts to the original source listed in package.json with every npm install and it still gows bigger with all the dependencies...

Reply:
hm.. not sure what's causing that ENOENT error...
i used this to write files
```js
async function writeFile(p, data) {
	return fsp.mkdir(path.dirname(p), { recursive: true })
		.then(x => fsp.writeFile(p, data));
}
```
maybe id doesn't work consistently across systems?

Reply:
also regarding npm link i think we should consider how boilerplate repo would be cloned to become a starting point for a bitsy game, rather than to work on the boilerplate functionality itself... i'm getting a couple of my bitsies to use boilerplate workflow and it seems to be more convenient to have a straightforward set of instructions in terms of setting everything up and to have a folder link that doesn't revert to the npm or github repo with new npm installs

Reply:
ah ok, my expected use for the boilerplate was to have a single clone of it and just copy hacks/gamedata in as needed; maybe it's better to avoid specifics in the instructions here since different people are going to have different workflows, and we can't realistically provide a tutorial for all of em at this level; could replace it with something more generic like
```
remember you can swap out the `bitsy-hacks` dependency with a custom version of it if needed
```
?

---

i didn't look too deep to see what the cause of the file error was, but usually when folks have to deal with nested directories, copying over existing stuff, etc they'll pull in something like `fs-extra` (https://www.npmjs.com/package/fs-extra) to simplify it; for example that same function here would just be
```
await fs.writeFile(p, data);
```
with `fs-extra` since it would automatically handle creating the directories if they don't exist + overwriting the file

if you can't repro when cloning a fresh copy i imagine it'll be a pain to debug, so lmk if you would rather i take a stab at fixing that bit

Reply:
added fs-extra since it sounds like a better solution anyways and also allows to get rid of some awkward code in index.mjs too... hopefully it will work now?

Reply:
i'm trying out npm-link workflow again to get better idea what wold be better to write in readme, and its just too awkward to use.. like after linking your local repo globablly and cloning boilerplate you would run `npm link @bitsy/hecks` before `npm i` so that it doesn't download unneccessary files, but then it would do this anyway, replace symlink with a new folder and download bitsy-hacks dependencies in boilerplate node-modules... so you can use `npm i && npm link @bitsy/hecks` but it would still download unnecessary deps and it is harder to keep track of. i don't see how it can ever be useful here, it's just so frustrating to work with.. i think it would be better to give one possible direction that works better in this case... maybe give some more notes to remind folks their package.json would point to the local file..

Reply:
the process for `link` is clone -> `npm i` -> `npm link <dependency>`; if by downloading unnecessary files you mean it causes it to download hacks from the npm registry before replacing them with a linky to your local ones, that happens regardless of whether you're linking or installing over `bitsy-hacks`, since it's just part of the initial `npm i`

regardless, the comment about providing one direction which works better is actually why i'm suggesting we leave it more general: what "works better" is going to be different depending on individual workflows (another way to get the same effect here would be to simply edit the copy in `node_modules/@bitsy/hecks` directly and someone unfamiliar with npm tooling may find that "better")